src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -168,16 +166,13 @@ internal Span<byte> GetSpan(int sizeHint)
                 AllocateWriteHeadUnsynchronized(sizeHint);
```

Comment:
Can put this in 
```csharp
if (_writingMemory.Length < Math.Max(1, sizeHint))
```
Would also need to move
```
_operationState.BeginWrite();
```
from `AllocateWriteHeadUnsynchronized` here (and to `GetMemory`)

Reply:
I'm not yet convinced this is right. _writingMemory would need to be updated with _writingHead.

Reply:
Would need lock free handlining on `_operationState` so probably out of scope (for reset in `AdvanceReader`)

Reply:
k something like?
```csharp
if (!_operationState.IsWritingActive ||
    _writingMemory.Length < Math.Max(1, sizeHint))
{
    lock (_sync)
    {
        AllocateWriteHeadUnsynchronized(sizeHint);
    }
}

return _writingMemory.Span;
```

Reply:
Which would work for the `BufferWriter` pattern where it runs out of space and does
```csharp
_output.Advance(_buffered);
_buffered = 0;
_span = _output.GetSpan(count);
```